src/ripple/app/paths/tests/Path_test.cpp

```diff
@@ -1,212 +1,585 @@
+        Env env(*this);
+        env.fund(XRP(10000), "alice", "bob");
+        env.trust(Account("bob")["USD"](1000), "alice");
+        env.trust(Account("alice")["USD"](1000), "bob");
+
+        Json::Value jv;
+        Json::Reader().parse(R"({
+                "Balance" : {
+                    "currency" : "USD",
+                    "issuer" : "rrrrrrrrrrrrrrrrrrrrBZbvji",
+                    "value" : "0"
+                },
+	            "Flags" : 196608,
+                "HighLimit" : {
+                    "currency" : "USD",
+                    "issuer" : "rPMh7Pi9ct699iZUTWaytJUoHcJ7cgyziK",
+                    "value" : "1000"
+                },
+                "HighNode" : "0000000000000000",
+                "LedgerEntryType" : "RippleState",
+                "LowLimit" : {
+                    "currency" : "USD",
+                    "issuer" : "rG1QQv2nh2gr7RCZ1P8YYcBUKCCN633jCn",
+                    "value" : "1000"
+                },
+                "LowNode" : "0000000000000000"
+            })", jv);
+
+        auto const jv_l = env.le(keylet::line(Account("bob").id(),
+            Account("alice")["USD"].issue()))->getJson(0);
+        for (auto it = jv.begin(); it != jv.end(); ++it)
+            expect(*it == jv_l[it.memberName()]);
+
+        env.trust(Account("bob")["USD"](0), "alice");
+        env.trust(Account("alice")["USD"](0), "bob");
+        expect(env.le(keylet::line(Account("bob").id(),
+            Account("alice")["USD"].issue())) == nullptr);
+    }
 
-public:
-    void run()
+    void
+    test_trust_auto_clear_trust_auto_clear()
+    {
+        testcase("trust auto clear");
+        Env env(*this);
+        env.fund(XRP(10000), "alice", "bob");
+        env.trust(Account("bob")["USD"](1000), "alice");
+
+        Json::Value jv;
+        Json::Reader().parse(R"({
+                "Balance" : {
+                    "currency" : "USD",
+                    "issuer" : "rrrrrrrrrrrrrrrrrrrrBZbvji",
+                    "value" : "0"
+                },
+	            "Flags" : 65536,
+                "HighLimit" : {
+                    "currency" : "USD",
+                    "issuer" : "rPMh7Pi9ct699iZUTWaytJUoHcJ7cgyziK",
+                    "value" : "0"
+                },
+                "HighNode" : "0000000000000000",
+                "LedgerEntryType" : "RippleState",
+                "LowLimit" : {
+                    "currency" : "USD",
+                    "issuer" : "rG1QQv2nh2gr7RCZ1P8YYcBUKCCN633jCn",
+                    "value" : "1000"
+                },
+            	"LowNode" : "0000000000000000"
+            })", jv);
+
+        auto const jv_l = env.le(keylet::line(Account("bob").id(),
+            Account("alice")["USD"].issue()))->getJson(0);
+        for (auto it = jv.begin(); it != jv.end(); ++it)
+            expect(*it == jv_l[it.memberName()]);
+
+        env(pay("bob", "alice", Account("bob")["USD"](50)));
+        env.trust(Account("bob")["USD"](0), "alice");
```

Comment:
It would be a bit more work here (constructing a second Json object with a balance and 0 limits), but the original test verified that the trustline still exists after alice clears trust, but before paying bob back.